frontend/widgets/OBSBasic_SceneItems.cpp

```diff
@@ -35,6 +35,29 @@
 
 using namespace std;
 
+namespace {
+std::string getNewSourceName(std::string_view name)
+{
+	std::string newName(name);
+
+	int suffix = 1;
+
+	for (;;) {
+		OBSSourceAutoRelease existing_source = obs_get_source_by_name(newName.c_str());
+		if (!existing_source) {
+			break;
+		}
+
+		newName = name;
+		newName += " (";
+		newName += std::to_string(suffix);
```

Comment:
See comment above, as-is this code would potentially automatically provide a fallback name like "My Name (1) (1)" because the suffix counter is never incremented at every iteration of the loop.

My conceptual dislike of this pattern also applies here.